### PR TITLE
Add a function field has_exception on stock.shipment.out

### PR DESCRIPTION
**Pull request by bhavana94**

### CODE REVIEW

File: shipment.py [hunk 1]
```diff
@@ -158,9 +158,13 @@ class ShipmentOut:
...
     )
 
+    has_exception = fields.Function(
+        fields.Boolean("Has Channel Expection"), 'get_has_channel_exception'
+    )
+
     weight = fields.Function(
         fields.Float(
-            "Weight", digits=(16,  Eval('weight_digits', 2)),
+            "Weight", digits=(16, Eval('weight_digits', 2)),
```
Comment (riteshshrv) [inline shipment.py]: revert

Comment (bhavana94) [inline shipment.py]: flake8 issue was there

File: shipment.py [hunk 2]
```diff
@@ -201,6 +205,20 @@ class ShipmentOut:
...
 
+    def get_has_channel_exception(self, name):
+        """
+        Returs True if sale has exception
+        """
+        ChannelException = Pool().get('channel.exception')
+
+        return bool(
+            ChannelException.search([
+                ('origin', '=', '%s,%s' % (self.__name__, self.id)),
```
Comment (riteshshrv) [inline shipment.py]: `self` won't work here; you need to give `name` and `id` of `sale` from where shipment was generated

File: shipment.py [hunk 3]
```diff
@@ -201,6 +205,22 @@ class ShipmentOut:
...
+        """
+        Returs True if sale has exception
+        """
+        ChannelException = Pool().get('channel.exception')
+        Sale = Pool().get('sale.sale')
+        SaleLine = Pool().get('sale.line')
+
+        return bool(
+            ChannelException.search([
+                ('origin', '=', '%s,%s' % (Sale.shipments, SaleLine.sale)),
```
Comment (riteshshrv) [inline shipment.py]: Wrong!
`Sale` and `SaleLine` both are class and what is this `SaleLine` doing here anyway?

File: shipment.py [hunk 4]
```diff
@@ -201,6 +205,21 @@ class ShipmentOut:
...
+    def get_has_channel_exception(self, name):
+        """
+        Returs True if sale has exception
+        """
+        ChannelException = Pool().get('channel.exception')
+        Sale = Pool().get('sale.sale')
+
+        return bool(
+            ChannelException.search([
+                ('origin', '=', '%s,%s' % (Sale.sale.id, Sale.sale.rec_name)),
```
Comment (riteshshrv) [inline shipment.py]: Wrong!
Sale and SaleLine both are classes!

File: shipment.py [hunk 5]
```diff
@@ -201,6 +205,15 @@ class ShipmentOut:
...
 
+    def get_has_exception(self, name):
+        """
+        Returs True if sale has exception
+        """
+        for sale in self.sales:
+            if sale.has_channel_exception:
+                return True
+            return False
+
```
Comment (riteshshrv) [inline shipment.py]: Wrong

**Comment by riteshshrv** [comment]
LGTM

**Pull request merged**